site/content/blog/2019-12-18-kubecon-na-2019.md

```diff
@@ -0,0 +1,70 @@
+
+Talking to other attendees, I found that what seemed like clear advantages of operator patterns vs. Helm charts often didn’t boil down to a substantial reason to choose operators in all cases. Much of this seemed to stem from a mixture of Helm being more well-known and understood, along with many operators not yet offering functionality that Helm didn’t already provide. I also noted that I approached the Helm vs operator conversation with something like management of a long-lived critical full-stack application in mind. When doing so, the advantages of an operator become much clearer. Knowing that your operator can handle something like a database schema migration check, or deploying your app with blue/green versioning, helps contrast the two options.
+
+This idea was presented well by some of the members of this operator panel: 
+
+{{< youtube zVyDYz69kaE >}}
+<br/>
+
+At the end of the day, Helm is still fulfilling a really important use-case in Kubernetes, but I believe this diagram shows how well-designed operators should eventually provide something truly unique.
+
+![Operator capability diagram](/blog/2019-12-18-kubecon-na-2019/capability-level-diagram.svg)
+
+#### Monomicroliths?
+
+One decision we constantly run up against when creating cloud infrastructure is whether it should be handled as a microservice or not. The core [cloud server](https://github.com/mattermost/mattermost-cloud) we are working on already has a lot of responsibility and continues to increase in complexity over time. Before we decide to add anything substantial to it, we start by questioning whether a microservice could fit our needs instead. Doing so allows us to seperate areas of security and responsibility in a way that also helps with horizontally scaling concerns; and has served us well so far.
+
+It goes without saying that we are just trying to follow an increasingly-common trend here. For this reason, I was happily surprised with the concepts presented by Goutham Veeramachaneni & Edward Welch in this talk:
+
+{{< youtube BmPnNmN9jtc >}}
+<br/>
+
+Their idea of a Monomicrolith really resonated. One of the struggles we have with the cloud server is that its responsibility to maintain our various cloud objects require it to be the source of truth in when creating, updating, or tearing down infrastructure. This, in turn, means that a lot of configuration is defined through the server. The talk explored the idea that this sort of design isn’t necessarily a flaw. Instead, if the code is modular enough, you can specify what pieces of functionality each process should run on startup. This allows for most of the benefits of a single code-base and configuration style, while still allowing for extreme flexibility and horizontal scaling.
```

Comment:
```suggestion
Their idea of a Monomicrolith really resonated. One of the struggles we have with the cloud server is that its responsibility to maintain our various cloud objects require it to be the source of truth when creating, updating, or tearing down infrastructure. This, in turn, means that a lot of configuration is defined through the server. The talk explored the idea that this sort of design isn’t necessarily a flaw. Instead, if the code is modular enough, you can specify what pieces of functionality each process should run on startup. This allows for most of the benefits of a single code-base and configuration style, while still allowing for extreme flexibility and horizontal scaling.
```

Above changes `require it to be the source of truth in when creating, updating,` to `require it to be the source of truth when creating, updating,`